File: 0633-sum-of-square-numbers/0633-sum-of-square-numbers.py
class Solution:
    def judgeSquareSum(self, c: int) -> bool:
        i = 0
        j = ceil(sqrt(c))
        
        while i <= j:
            sum = i*i + j*j
            if sum < c:
                i += 1
            elif sum > c:
                j -= 1
            else:
                return True
        return False

File: 0633-sum-of-square-numbers/0633-sum-of-square-numbers.js
/**
 * @param {number} c
 * @return {boolean}
 */
var judgeSquareSum = function(c) {
    if (c === 0) {
        return true
    }
    for (let a = 0; a*a < c; a++) {
        let b = Math.sqrt(parseFloat(c-a*a));
        if (b - Math.round(b) === 0) {
            return true;
        }
    }
        return false
};